snuba/subscriptions/data.py

```diff
@@ -229,8 +243,17 @@ def add_conditions(
                 Literal(None, timestamp),
             ),
         ]
-
-        if offset is not None:
+        if self.organization:
```

Comment:
I know this is something that was already like this before your PR but personally I don't think we should be putting code that can handle every single dataset here. It will get more convoluted as we add more datasets. If building the SubscriptionData is specific for each entity, then why not  define a function on the entity itself that knows how to build this and just call the relevant function for the entity we are concerned with from here.

Reply:
I don't think we want to encode knowledge in the entity about how it's being used (subscriptions vs regular queries). Also I don't think this is dataset specific, if any dataset sends us an organization we will correctly add it to the conditions. That doesn't seem wrong.

Reply:
This is why I think it's dataset specific:

- `organization_id` is relevant for sessions but not for events or transactions
- `offset` is the same, relevant for events and transactions not sessoins
- Today `project_id` is the same on the datasets but it seems plausible that some datasets support cross product queries in future and not others

Yes there are a few fields that might overlap but ultimately this just seems a co-incidence to me and not something that we should enforce. Basically every time we need to add a subscription for a dataset with slightly different fields we would need to change the function that gets run on *every* dataset rather than the part only specific to that kind of subscription and that doesn't seem right to me.